Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/ExternalPaymentMethodTokenizationViewModel.swift

```diff
@@ -698,14 +704,27 @@ extension ExternalPaymentMethodTokenizationViewModel: SFSafariViewControllerDele
             self.didPresentExternalView?()
         }
     }
+}
+
+extension ExternalPaymentMethodTokenizationViewModel {
     
+    private func unselectPaymentMethodWithError(_ error: Error) {
```

Comment:
I am not sure why you need a `ClientSession.Action.unselectPaymentMethod()` function in each view model, since it's never going to do anything different. Also, having it as a promise (as is in the `ClientSession.Action`) is useful cause you can always use it as the step 1 in `firstly`.

Reply:
Addressed ✅